argilla/docs/how_to_guides/migrate_from_legacy_datasets.md

```diff
@@ -1 +1,287 @@
+        """ This function maps a text2text record dictionary to the new Argilla record."""
+        suggestions = []
+        responses = []
+    
+        if prediction := data.get("prediction"):
+            first = prediction[0]
+            agent = data["prediction_agent"]
+            suggestions.append(
+                rg.Suggestion(question_name="text_generation", value=first["text"], score=first["score"], agent=agent)
+            )
+    
+        if annotation := data.get("annotation"):
+            # From data[annotation]
+            user_id = users_by_name.get(data["annotation_agent"], current_user).id
+            responses.append(rg.Response(question_name="text_generation", value=annotation, user_id=user_id))
+    
+        vectors = (data.get("vectors") or {})
+        return rg.Record(
+            id=data["id"],
+            fields={"text": data["text"]},
+            # The inputs field should be a dictionary with the same keys as the `fields` in the settings
+            metadata=data["metadata"],
+            # The metadata field should be a dictionary with the same keys as the `metadata` in the settings
+            vectors=[rg.Vector(name=name, values=value) for name, value in vectors.items()],
+            # The vectors field should be a dictionary with the same keys as the `vectors` in the settings
+            suggestions=suggestions,
+            responses=responses,
+        )
+    ```
+
+The functions above depend on the `users_by_name` dictionary and the `current_user` object to assign responses to users, we need to load the existing users. You can retrieve the users from the Argilla V2 server and the current user as follows:
+
+```python
+# For 
+users_by_name = {user.username: user for user in client.users}
+current_user = client.me
+```
+
+Finally, upload the records to the new dataset using the `log` method and map functions.
+
+```python
+records = []
+
+for data in hf_records:
+    records.append(map_to_record_for_single_label(data, users_by_name, current_user))
+
+# Upload the records to the new dataset
+dataset.records.log(records)
+```
+You have now successfully migrated your legacy dataset to Argilla V2. For more guides on how to use the Argilla SDK, please refer to the [How to guides](index.md).
+
+## Migrating Feedback Datasets on your Argilla 1.* server
+
+As mentioned above, `FeedbackDataset`'s are compatible with Argilla V2 and do not need to be reformatted. However, you may want to migrate your feedback datasets to the new server so that you can deprecate your Argilla 1.* server. Here is a guide on how to migrate your feedback datasets:
+
+```python
+import argilla.v1 as rg_v1
+
+# Initialize the API with an Argilla server less than 2.0
+old_client = rg.Argilla(old_server_api_url, old_server_api_key)
+new_client = rg.Argilla(new_server_api_url, new_server_api_key) 
+
+dataset_name = "feedback-dataset"
+old_dataset = old_client.datasets(name=dataset_name)
+new_dataset = new_client.datasets.add(dataset)
```

Comment:
I guess this line should be `new_dataset = new_client.datasets.add(old_dataset)`? But it still throws an error even with the change.